cmd/syncthing/gui.go

```diff
@@ -188,16 +210,15 @@ func startGUI(cfg config.GUIConfiguration, assetDir string, m *model.Model) erro
 		ReadTimeout: 10 * time.Second,
 	}
 
-	csrv := &folderSummarySvc{model: m}
-	go csrv.Serve()
+	s.fss.ServeBackground()
 
-	go func() {
-		err := srv.Serve(listener)
-		if err != nil {
-			panic(err)
-		}
-	}()
-	return nil
+	err := srv.Serve(s.listener)
+	l.Warnln("API:", err)
```

Comment:
I looked at it, but we'll get a "accept on closed socket" or something even in the graceful case, and closing the listening socket is the only graceful way of interrupting `http.Serve()`. Also `suture` will add a whole bunch of log output when restarting the service, which may be something we need to manage...